Packages/vcs/Lib/vtk_ui/button.py

```diff
@@ -135,12 +135,20 @@ def add_state(self, label=None, image=None, bgcolor=None, fgcolor=None, opacity=
         self.states.append(ButtonState(label=label, image=image, bgcolor=bgcolor, fgcolor=fgcolor, opacity=opacity))
 
     def place(self):
+
         width, height = self.get_dimensions()
         x, y = self.get_position()
         bounds = (x, x + width, y - height, y, 0, 0)
-
         self.repr.SetPlaceFactor(1)
         self.repr.PlaceWidget(bounds)
+        self.repr.Modified()
+        if self.showing():
+            # This One Weird Hack will make your Buttons Go In the Right Place - Developers hate it!
+            # Buttons weren't always getting properly placed (toolbars in toolbars being the canonical example)
+            # This makes them show up correctly. Weird, but it works.
+            h_state = self.repr.GetHighlightState()
+            self.repr.Highlight((h_state + 1) % 3)
+            self.repr.Highlight(h_state)
```

Comment:
Yeah, you're telling me. I actually have a tendency to delve down into the C++ source when I find a weird hack like this, so I can ideally *not* have the weird hack anymore. This time I failed.